### DISCUSSION TITLE
Referencing custom sphinx roles from myst_parser

### DISCUSSION THREAD

**Discussion by nikomatsakis** [executablebooks/MyST-Parser#988] [Q&A] [answered]
Hi folks! I am trying to extend an [existing Sphinx project](https://github.com/nikomatsakis/fls) based on restructuredtext to make use of `msyt_parser`. This project uses some [custom roles and things](https://github.com/nikomatsakis/fls/blob/301d33b12caaa125f4b7e7151837660e248458b7/src/conf.py#L22-L30). I've modified the list of extensions to include myst:

```py
extensions = [
    "myst_parser",
    "ferrocene_spec",
    "ferrocene_spec_lints",
    "ferrocene_toctrees",
]
```

but when I try to build [markdown files that reference a custom role](https://github.com/nikomatsakis/fls/blob/main/src/macros-by-example.md) (e.g., `syntax`), I get [errors](https://github.com/nikomatsakis/fls/actions/runs/11503203661/job/32019957700#step:5:140):

```
macros-by-example.md:3: WARNING: Unknown directive type: 'syntax' [myst.directive_unknown]
```

Any idea what's going on?

Thanks!

**Comment by nikomatsakis** [comment]
(Also, I find it hard to believe I'm the first person having trouble with this, but my attempts to search did not find previous discussions. Apologies if I missed something obvious!)

**Comment by chrisjsewell** [comment, marked as answer]
`syntax` is under the `spec` domain, you should be using `spec:syntax`

**Reply by nikomatsakis** [reply]
Aha! Thanks!

**Comment by nikomatsakis** [comment]
@chrisjsewell is there a way to set the "Default domain" in markdown files?

**Reply by chrisjsewell** [reply]
Have you tried using the default-domain directive, like you do in the rst files?

**Reply by nikomatsakis** [reply]
No :) I am asking what the syntax for that is. I suppose I can RTFM, however.

**Reply by nikomatsakis** [reply]
Thanks again!

**Reply by chrisjsewell** [reply]
Just the same as any directive, like you used `syntax`
